classes/local/store/object_file_system.php

```diff
@@ -415,21 +419,25 @@ public function get_content(\stored_file $file) {
      * @throws \coding_exception
      */
     public function xsendfile_file(stored_file $file): bool {
-        $contenthash = $file->get_contenthash();
-        if ($this->presigned_url_configured() &&
+        if (!$this->is_configured()) {
+            return parent::xsendfile_file($file);
+        } else {
```

Comment:
why not just ditch the else completely? we exit above. same for the other methods

Reply:
Done